app/controllers/kudos_controller.rb

```diff
@@ -1,22 +1,31 @@
 class KudosController < ApplicationController
   before_action :authenticate_employee!
-  before_action :set_kudo, only: %i[show edit update destroy]
 
   def index
-    @kudos = Kudo.all
+    @kudos = Kudo.all.includes(:company_value, :receiver, :giver)
   end
 
-  def show; end
+  def show
+    set_kudo
+  end
 
   def new
     @kudo = Kudo.new
+    @available_employees = Employee.where.not(id: current_employee.id)
+    @company_values = CompanyValue.all
   end
 
-  def edit; end
+  def edit
+    set_kudo
+    @available_employees = Employee.where.not(id: current_employee.id)
+    @company_values = CompanyValue.all
+  end
 
   def create
     @kudo = Kudo.new(kudo_params)
     @kudo.giver = current_employee
+    @available_employees = Employee.where.not(id: current_employee.id)
```

Comment:
do we really need @available_emplyess here? I would consider adding another check, that giver is different from receiver here

Reply:
I have validate method in model of Kudo about that giver is different from receiver. But in this case when we don`t save kudo we need this variable for render :new. I put all variables of form in private method in kudos_controller as form_params method, it is good practice?